ethereum/contracts/relayer/wormholeRelayer/WormholeRelayerDelivery.sol

```diff
@@ -282,19 +269,22 @@ abstract contract WormholeRelayerDelivery is WormholeRelayerBase, IWormholeRelay
         DeliveryResults memory results;
 
         // Enforce replay protection
-        if (getDeliverySuccessState().deliverySuccessBlock[vaaInfo.deliveryVaaHash] != 0) {
+        mapping(bytes32 => uint256) storage deliverySuccessBlock =  getDeliverySuccessState().deliverySuccessBlock;
+        if (deliverySuccessBlock[vaaInfo.deliveryVaaHash] != 0) {
             results.status = DeliveryStatus.RECEIVER_FAILURE;
             results.additionalStatusInfo = "Delivery already executed";
             setDeliveryBlock(results.status, vaaInfo.deliveryVaaHash);
             emitDeliveryEvent(vaaInfo, results);
             return;
         }
+        // Optimisitcally set to also prevent reentrancy
+        deliverySuccessBlock[vaaInfo.deliveryVaaHash] = block.number;
+
 
         // Forces external call
         // In order to catch reverts
-        // (If the user's contract requests a forward
-        // and there ends up not being enough funds,
-        // then we will revert this call)
+        // (Previously needed for reverts, but waiting to remove since this is sensitive code 
+        //  and would need a real audit to redeploy)
         try 
```

Comment:
correct me if I'm wrong, but executeInstruction should never revert right? Can we not remove the failure case of this

Reply:
as in, can we just change this into a normal function call rather than an external call